### PR TITLE
Update PropTypes for React v15.5

### PR DESCRIPTION
**Pull request by GruFFix**
Updated `import React, { PropTypes } from 'react';` to `import PropTypes from 'prop-types';`
and updated React -> v15.5.4

info: https://facebook.github.io/react/blog/2017/04/07/react-v15.5.0.html

### CODE REVIEW

**Comment by GruFFix** [comment]
@qimingweng can you look my pr please ?

**Comment by stevewillard** [comment]
Any reason why this can't be merged?

File: package.json [hunk 1]
```diff
@@ -24,11 +24,12 @@
     "keycode": "^2.1.0",
     "lodash": "^3.9.3",
     "narcissus": "^1.0.0",
+    "prop-types": "^15.5.8",
```
Comment (chhsiao1981) [inline package.json:27]: prop-types should be in peerDependencies?

**Comment by natenorberg** [comment]
Has anyone checked if this can be merged? It'd be nice to upgrade to fiber without having to rewrite this component: http://isfiberreadyyet.com/

**Comment by cordoval** [comment]
I am trying to use this fork @GruFFix 

with 

```json
"react-modal-dialog": "git@github.com:GruFFix/react-modal-dialog#c6ce77c1676790c1228716b894d5f3f3ccf25ea4",
```

but when i require it
```js
import { ModalContainer, ModalDialog } from 'react-modal-dialog'
```

i get 

```
ERROR in ./frontend/app/components/Modal.js
Module not found: Error: Can't resolve 'react-modal-dialog' in '/system/frontend/app/components'
 @ ./frontend/app/components/Modal.js 17:24-53
```
trying to understand why it is not resolving, if the import gets modified with 'react-modal-dialog/src' it gives problems for other subsequent files that need a loader.

**Comment by cordoval** [comment]
I made my own fork for now, hope to see this merged ASAP 🥇 👍

**Comment by GruFFix** [comment]
@cordoval  This error does exist. Module is not reassembled as for import package

**Comment by cordoval** [comment]
yeah correct, i have fixed that by doing a fork, did not want to complicate my deploy scripts
thanks for your reply

**Comment by hlolli** [comment]
Merge? Where are the maintainers :(

**Comment by colesimmons** [comment]
Please merge. This warning is driving me crazy.

**Comment by cordoval** [comment]
guys do not ask for this, what i did instead was creating my own modal with react-portal, so much way better and more secure for those using CSP, fully recommended. No need for this package, i removed it from package.json

**Review by sametcodes — APPROVED** [review]

**Pull request closed without merging**